components/generic/CompareImageSlider.vue

```diff
@@ -0,0 +1,225 @@
+<template>
+  <figure
+    ref="container"
+    class="compare-image"
+  >
+    <img
+      ref="imageLeft"
+      :src="imageLeft"
+      :alt="imageLeftText"
+      class="compare-image__image-left"
+      :style="leftImageClip"
+    >
+    <img
+      ref="imageRight"
+      :src="imageRight"
+      :alt="imageRightText"
+      class="compare-image__image-right"
+    >
+    <div
+      ref="slider"
+      class="compare-image__slider"
+      :style="sliderBarPosition"
+    >
+      <span class="compare-image__slider__bar" />
+      <button
+        :class="{ 'is-active' : dragging }"
+        class="compare-image__slider__handle"
+      >
+        <span class="sr-only">Slider Handle</span>
+      </button>
+    </div>
+  </figure>
+</template>
+
+<script>
+  export default {
+    name: 'CompareImageSlider',
+
+    props: {
+      imageLeft: {
+        type: String,
+        required: true
+      },
+
+      imageRight: {
+        type: String,
+        required: true
+      },
+
+      imageLeftText: {
+        type: String,
+        default: 'This is an image'
+      },
+
+      imageRightText: {
+        type: String,
+        default: 'This is an image'
+      }
+    },
+
+    data() {
+      return {
+        dragging: false,
+        imageWidth: 0,
+        sliderWidth: 0,
+        sliderPosition: 0.5
+      };
+    },
+
+    computed: {
+      leftImageClip() {
+        return {
+          clip: `rect(auto, ${this.imageWidth * this.sliderPosition}px, auto, auto)`
+        };
+      },
+
+      sliderBarPosition() {
+        return {
+          left: `${this.imageWidth * this.sliderPosition - this.sliderWidth / 2}px`
```

Comment:
Let's move the calculation out into a `const` and just have that in the `${}`, to improve readability.